File: 3-Python/1-Programas/1-EstructurasSimples/ejercicio2.py
consumo = float(input("Ingrese consumo en kWh: "))
tarifa = int(input("Tipo de tarifa (1=Básica, 2=Intermedia, 3=Alta): "))

costo_kwh = 0.0
recargo_pct = 0.0

if tarifa == 1:
    costo_kwh = 0.85
    if consumo > 250: recargo_pct = 0.12
elif tarifa == 2:
    costo_kwh = 1.25
    if consumo > 500: 
        recargo_pct = 0.18
    elif consumo >= 300: 
        recargo_pct = 0.10
elif tarifa == 3:
    costo_kwh = 2.10
    if consumo > 400: recargo_pct = 0.25

subtotal = consumo * costo_kwh
monto_recargo = subtotal * recargo_pct
total = subtotal + monto_recargo

mensaje = ""
if total > 1500: mensaje = "Consumo elevado: se recomienda ahorro de energía" 
if consumo < 100: mensaje = "Usuario con consumo eficiente"

print(f"--- Factura ---\nCosto por kWh: ${costo_kwh}")
print(f"Recargo aplicado: ${monto_recargo:.2f}")
print(f"Total a pagar: ${total:.2f}")
if mensaje: print(mensaje)

File: 2-Java/1-Programas/2-EstructurasControl/1-Actividad De Desarrollo 1. Ejercicios de Estructuras de Control Selectivas/Ejercicio2.java
import javax.swing.JOptionPane;

public class Ejercicio2 {
    public static void main(String[] args) {
        double consumo = Double.parseDouble
        (JOptionPane.showInputDialog("Ingrese consumo mensual (kWh):"));
        int tarifa = Integer.parseInt
        (JOptionPane.showInputDialog("Seleccione Tarifa (1=Básica, 2=Intermedia, 3=Alta):"));

        double costoKwh = 0;
        double recargoPorcentaje = 0;
        switch (tarifa) {
            case 1:
                costoKwh = 0.85;
                if (consumo > 250) recargoPorcentaje = 0.12;
                break;
            case 2:
                costoKwh = 1.25;
                if (consumo > 500) recargoPorcentaje = 0.18;
                else if (consumo >= 300) recargoPorcentaje = 0.10;
                break;
            case 3:
                costoKwh = 2.10;
                if (consumo > 400) recargoPorcentaje = 0.25;
                break;
            default:
                JOptionPane.showMessageDialog(null, "Tarifa inválida");
                return;
        }
        double subtotal = consumo * costoKwh;
        double montoRecargo = subtotal * recargoPorcentaje;
        double total = subtotal + montoRecargo;

        String mensaje = "";
        if (total > 1500) mensaje = "Consumo elevado: se recomienda ahorro de energía";
        if (consumo < 100) mensaje = "Usuario con consumo eficiente";

        JOptionPane.showMessageDialog(null, 
            "Tarifa: " + tarifa +
            "\nConsumo: " + consumo + " kWh" +
            "\nCosto por kWh: $" + costoKwh +
            "\nRecargos: $" + montoRecargo +
            "\nTotal a pagar: $" + total +
            "\n" + mensaje);
    }
}